frontend/src/components/authentication/ForgotPassword.tsx

```diff
@@ -0,0 +1,114 @@
+import React, {useState, useCallback} from 'react'
+import {CssBaseline, TextField} from '@material-ui/core'
+import {Button, Avatar, Typography, Link, Paper, Grid, Box }  from '@material-ui/core'
+import LockOutlinedIcon from '@material-ui/icons/LockOutlined'
+
+import * as gridSize from '../../constants/styles.values'
+import {SIX as SHADOW_DEPTH_SIX} from '../../constants/styles.values'
+import { emailFormatIsValid } from '../../utils/FormatVerifications'
+import {useStyles} from './SignUp.styles'
+import { AuthApiClient } from '../../api/authRequest'
+
+export default function ForgotPasswordSide(): React.ReactElement {
+
+  const classes = useStyles()
+
+  const initialState = {
+    email: '',
+    message: '',
+    error: false
+  }
+
+  const [state, setState] = useState(initialState)
+  
+  const onChangeEmail = useCallback((e) => {
+    setState({...state, email: e.target.value})
+  }, [setState, state])
+  
+  const recoverPassword = useCallback((e) => {
+    e.preventDefault()
+    if (emailFormatIsValid(state.email)) {
+      const authApiClient = new AuthApiClient()
+      authApiClient.recover(state.email).then((res) => {
+        setState({...state, message: res.message, error: res.error})
+      })
+      .catch((err) => {
+        setState({...state, message: err.message, error: true})
+      })
+    } else {
+      setState({...state, message: 'Invalid e-mail format.', error: true})
+    }
+  }, [setState, state])
```

Comment:
You don't need to provide `setState` method in dependencies.

Reply:
Ok